google/cloud/pubsublite/internal/resumable_async_streaming_read_write_rpc_test.cc

```diff
@@ -0,0 +1,462 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsublite/internal/resumable_async_streaming_read_write_rpc.h"
+#include "google/cloud/pubsublite/testing/mock_async_reader_writer.h"
+#include "google/cloud/pubsublite/testing/mock_backoff_policy.h"
+#include "google/cloud/pubsublite/testing/mock_retry_policy.h"
+#include "google/cloud/future.h"
+#include "google/cloud/status_or.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include "absl/memory/memory.h"
+#include <gmock/gmock.h>
+#include <chrono>
+#include <deque>
+#include <memory>
+
+namespace google {
+namespace cloud {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace pubsublite_internal {
+namespace {
+
+using ::google::cloud::testing_util::IsOk;
+using ::testing::ByMove;
+using ::testing::InSequence;
+using ::testing::MockFunction;
+using ::testing::Return;
+using ::testing::StrictMock;
+
+using ::google::cloud::pubsublite_testing::MockBackoffPolicy;
+using ::google::cloud::pubsublite_testing::MockRetryPolicy;
+
+using ms = std::chrono::milliseconds;
+const unsigned int kFutureWaitMs = 100;
```

Comment:
consider:

```
auto constexpr kFutureWait = std::chrono::milliseconds(100);
```

But I wonder why are you introducing the delay?  Is waiting for 100 milliseconds vs. 50 or 10 or 0 any different in these tests?

Reply:
the magic number used for the delay is never actually used. Regardless of the value, the sleeper mock in the test suite either returns an immediate future with zero time delay or returns the future of a promise that's satisfied manually later

Reply:
I do not think that is accurate, see: https://github.com/googleapis/google-cloud-cpp/pull/8450/files#r814918494